hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -62,4 +63,7 @@
      * @return sampled entries.
      */
     Iterable<LazyEntryViewFromRecord> getRandomSamples(int sampleCount);
+
+    MapKeyIteratorResult fetchNext(int tableIndex, int size);
+
```

Comment:
To me `fetch` would be a better name. I know that it returns the next `tableIndex` but it doesn't fetch from the nextIndex, but the given one, so it's a bit misleading to me.